### PR TITLE
Make it possible to add a list of monsters to NOT trigger alarm.

### PR DESCRIPTION
**Pull request by fosJoddie**
## Description
Implements a monster blacklist that will make filters fail for certain pokemon.

Example use: 

Alert about any pokemon above 15/14/14 iv, unless they are certain "trash" pokemon:

```
"f_15_14_14_ignore_trash":{
"ignore_monsters": [ 10,11,12,13,14,15,16,17,18,19,20,21,22,23,24,27 ... ],
   "min_atk": 15, "max_atk": 15,
   "min_def": 14, "max_def": 15,
   "min_sta": 14, "max_sta": 15,
   "min_iv": 0.0, "max_iv": 100, 
   "is_missing_info": false
}
```

## Type of Change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (would cause existing functionality to change)

## Motivation and Context
Instead of writing a huge list of pokemon I WANT alerts on, I wanna write a smaller list of pokemon I do not want.

## How Has This Been Tested?
I wrote a python parser on the bathroom stall at work and ran my code through it there. It seems to work fine until it runs out of toilet paper.

## Wiki Update
Do you still use the wiki? I wrote up the change in the monster filter docs file
- [x] This change requires an update to the Wiki.
- [ ] This change does not require an update to the Wiki.

## Checklist
- [x] This change follows the code style of this project.
- [x] This change only changes what is necessary to the feature.

### CODE REVIEW

**Comment by billyjbryant** [comment]
What happens if you have both a `"monsters"` list and an `"ignore_monsters"` list in an incident that they conflict with each other?

**Comment by billyjbryant** [comment]
You should also look at creating the necessary tests within the unit testing, this will prove that the filters work as expected and satisfies the "How Has This Been Tested?" requirement of the PR

**Comment by fosJoddie** [comment]
If something is in the ignore filter, it will fail the filter for that pokemon. 
Blacklist takes precedence over Whitelist in this case, simply because of the way the filter logic works

File: tests/filters/test_monster_filter.py [hunk 1]
```diff
@@ -41,6 +41,31 @@ def test_monster_id(self):
         for e in [fail1, fail2, fail3]:
             self.assertFalse(mon_filter.check_event(e))
 
+    def test_ignore_monster_id(self):
+        # Create the filters
+        settings = {
+            "monsters": [1, "2", "Venusaur"],
```
Comment (billyjbryant) [inline tests/filters/test_monster_filter.py]: For testing here, I would remove `"monsters"` as we truly just want to test the ignore filter in this particular scenario.

Comment (fosJoddie) [reply tests/filters/test_monster_filter.py]: sure thing, updated

**Comment by kvangent** [comment]
Do we need to add a similar filter to raids?

**Comment by wickles** [comment]
What is the use case for this? Temporary ignore lists for events maybe?

**Comment by kvangent** [comment]
@wickles it's a blacklist for monsters

**Comment by darkelement1987** [comment]
I liked it the  way it was on older version of pokealarm (true/false)

**Pull request merged**